### DISCUSSION TITLE
How do get the last 100 messages from a topic?

### DISCUSSION THREAD

**Discussion by mskydream** [IBM/sarama#2506] [Q&A]
I am writing a separate function that accepts a broker and a topic. Is there such a possibility in the sarama library? If there is, can you help me?

**Comment by dnwe** [comment]
It's not obvious what "last 100 messages" would mean in the context of a topic containing multiple partitions 

However, if you're just talking about a single topic partition, then yes you should be able to attach a consumer to that specific partition, fetch the end offset of the log and then seek to 100 messages before that and consume them

**Comment by mskydream** [comment]
> It's not obvious what "last 100 messages" would mean in the context of a topic containing multiple partitions
> 
> However, if you're just talking about a single topic partition, then yes you should be able to attach a consumer to that specific partition, fetch the end offset of the log and then seek to 100 messages before that and consume them

Yes, I meant what you wrote. I do not know how to implement it. Can you write an example or send similar examples?
